File: algorithms/609. Find Duplicate File in System/solution.py
class Solution(object):
    def findDuplicate(self, paths):
        """
        :type paths: List[str]
        :rtype: List[List[str]]
        """
        group = collections.defaultdict(list)
        for p in paths:
            p = p.split()
            for i in xrange(1, len(p)):
                filename, content = p[i].split('(')
                group[content[:-1]].append(p[0] + '/' + filename)
        return [p for c, p in group.iteritems() if len(p) > 1]

File: algorithms/609. Find Duplicate File in System/solution.cc
class Solution {
public:
    vector<vector<string>> findDuplicate(vector<string>& paths) {
        unordered_map<string, vector<string>> group;
        for ( string & s : paths ) {
            istringstream iss(s);
            vector<string> tokens(istream_iterator<string> {iss}, istream_iterator<string> {});
            for ( int i = 1; i < tokens.size(); ++i ) {
                int p = tokens[i].find('(');
                string filename = tokens[i].substr(0, p);
                string content = tokens[i].substr(p + 1, tokens[i].size() - p - 2);
                group[content].push_back(tokens[0] + "/" + filename);
            }
        }
        vector<vector<string>> result;
        for ( auto & p : group ) {
            if ( p.second.size() > 1 ) {
                result.push_back(p.second);
            }
        }
        return result;
    }
};
